src/js/adc.js

```diff
@@ -17,42 +17,90 @@ var util = require('util');
 var adc = process.binding(process.binding.adc);
 
 
-// ADC(pin[, callback])
-function ADC(pin, callback) {
-  if (!util.isNumber(pin)) {
-    throw new TypeError('Bad arguments - pin');
+function Adc() {
+  if (!(this instanceof Adc)) {
+    return new Adc();
   }
+}
+
+Adc.prototype.open = function(configuration, callback) {
+  return new AdcPin(configuration, callback);
+};
 
-  this._pin = pin;
 
-  adc.export(this._pin, function(err) {
-    if (util.isFunction(callback)) {
-      callback(err);
+function AdcPin(configuration, callback) {
+  var self = this;
+
+  if (util.isObject(configuration)) {
+    if (process.platform === 'linux') {
+      if (!util.isString(configuration.device)) {
+        throw new TypeError(
+          'Bad configuration - device is mandatory and String');
```

Comment:
I think error message should be like `device is mandatory and should be String`. Current message right now a little bit confuses me that device is now type of String.